components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/CustomTabWindowFeature.kt

```diff
@@ -57,16 +57,29 @@ class CustomTabWindowFeature(
     }
 
     /**
-     * Starts the feature and a observer to listen for window requests.
+     * Starts observing the configured session to listen for window requests.
      */
     override fun start() {
-        sessionManager.findSessionById(sessionId)?.register(windowObserver)
+        scope = store.flowScoped { flow ->
+            flow.mapNotNull { state -> state.findCustomTab(sessionId) }
+                .ifChanged {
+                    it.content.windowRequest
+                }
+                .collect { state ->
+                    val windowRequest = state.content.windowRequest
+                    if (windowRequest?.type == WindowRequest.Type.OPEN) {
+                        val intent = configToIntent(state.config)
+                        intent.launchUrl(context, windowRequest.url.toUri())
+                        store.dispatch(ContentAction.ConsumeWindowRequestAction(sessionId))
```

Comment:
If some of the state changes may be dropped due to conflation, this `action` isn't guaranteed to be noticed by the code above... But I think that's probably fine - the reducer, and I'm assuming it will process _every_ action... -  will set the `windowRequest` value to `null`... So some future `UpdateWindowRequestAction` will reset `state.windowRequest` to a brand new instance...